src/Lifecycle.ts

```diff
@@ -1105,6 +1106,9 @@ export async function onLoggedOut(): Promise<void> {
  */
 async function clearStorage(opts?: { deleteEverything?: boolean }): Promise<void> {
     if (window.localStorage) {
+        // save the language code, if it has already been set
+        const language = SettingsStore.getValue("language");
```

Comment:
This comment doesn't align with what it does, as getValue will fall through to defaults unless `excludeDefaults` is specified

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/61551df5-a895-4322-86af-53a9170a9363)

This is a fresh incognito window where I did not set language for example.